iree/compiler/Dialect/HAL/Target/VulkanSPIRV/VulkanSPIRVTarget.cpp

```diff
@@ -108,6 +109,108 @@ static void recordFullExecutionBarrier(Value commandBuffer, Location loc,
       ArrayRef<Value>{memoryBarrier}, ArrayRef<Value>{});
 }
 
+/// Generates IR to compute the ceil(`numerator`, `denominator`).
+static Value computeCeil(Location loc, Value one, Value numerator,
+                         Value denominator, OpBuilder &builder) {
+  Value dm1 = builder.create<SubIOp>(loc, denominator, one);
+  return builder.create<SignedDivIOp>(
+      loc, builder.create<AddIOp>(loc, numerator, dm1), denominator);
+}
+
+/// Calculates the number of workgroups to use based on the shape of the result
+/// of the dispatch region, assuming there is a single result value, or all
```

Comment:
This is actually a very important assumption to hold the dispatch true.. Can we put it around the call site instead of attaching it here? This is just a normal utility function and it should not concern single vs. multiple results.

Reply:
It is at the call site. It is just added here as a comment to signify when this method is to be used. Not sure I follow the suggestion here actually.

Reply:
Tthis function is called by `calculateSPIRVDispatchWorkgroupCount`, which is then called by `recordDispatch`. I'd think the assumption should be put at Line 327 of this file to be clear? Here it's already nested in two function calls and this function does not concern multiple results at all.

Reply:
Updated comments and added assumption at 327.